### PR TITLE
Upgrade OndaEDFSchemas to Legolas 0.5, Onda 0.15

### PR DESCRIPTION
**Pull request by christopher-dG**
this is half of #60

### CODE REVIEW

File: OndaEDFSchemas.jl/test/runtests.jl [hunk 1]
```diff
@@ -7,61 +7,73 @@ using StableRNGs
...
-    tbl = Arrow.Table(Arrow.tobuffer(plans))
-    @test isequal(Tables.columntable(tbl), Tables.columntable(plans))
-end
+for v in [1, 2]
+    @testset "ondaedf.plan@$v" begin
+        rng = StableRNG(10)
+        plans = mock_plan(30; v, rng)
+        schema = Tables.schema(plans)
+        @test nothing === Legolas.validate(schema, Legolas.SchemaVersion("ondaedf.plan", v))
+        tbl = Arrow.Table(Arrow.tobuffer(plans; maxdepth=9))
```
Comment (kleinschmidt) [inline OndaEDFSchemas.jl/test/runtests.jl:65]: it's a bit odd that this is now necessary (even for the V1 schema)...can you maybe look through the CI runs to find the last time this ran successfully.  is there some reason why Legolas 0.5 would have changed something here?

File: OndaEDFSchemas.jl/test/runtests.jl [hunk 2]
```diff
@@ -7,61 +7,73 @@ using StableRNGs
...
 end
 
-@testset "ondaedf.plan@1" begin
-    rng = StableRNG(10)
-    plans = mock_plan(30; rng)
-    @test nothing === Legolas.validate(plans, Legolas.Schema("ondaedf.plan@1"))
-    tbl = Arrow.Table(Arrow.tobuffer(plans))
-    @test isequal(Tables.columntable(tbl), Tables.columntable(plans))
-end
+for v in [1, 2]
```
Comment (kleinschmidt) [inline OndaEDFSchemas.jl/test/runtests.jl:59]: you _could_ do this as a testset if you wanted...

```suggestion
@testset "Schema version $v" for v in (1, 2)
```

File: OndaEDFSchemas.jl/src/OndaEDFSchemas.jl [hunk 3]
```diff
@@ -1,94 +1,135 @@
...
+    # EDF.FileHeader field
+    seconds_per_record::Float64
+    # Onda.SamplesInfo fields (channels -> channel), may be missing
+    sensor_type::Union{Missing,AbstractString} = lift(String, sensor_type)
+    channel::Union{Missing,AbstractString} = lift(String, channel)
+    sample_unit::Union{Missing,AbstractString} = lift(String, sample_unit)
+    sample_resolution_in_unit::Union{Missing,LPCM_SAMPLE_TYPE_UNION} = lift(convert_number_to_lpcm_sample_type, sample_resolution_in_unit)
+    sample_offset_in_unit::Union{Missing,LPCM_SAMPLE_TYPE_UNION} = lift(convert_number_to_lpcm_sample_type, sample_offset_in_unit)
+    sample_type::Union{Missing,AbstractString} = lift(onda_sample_type_from_julia_type, sample_type)
+    sample_rate::Union{Missing,LPCM_SAMPLE_TYPE_UNION} = lift(convert_number_to_lpcm_sample_type, sample_rate)
```
Comment (kleinschmidt) [inline OndaEDFSchemas.jl/src/OndaEDFSchemas.jl:55]: I think these fields have different types now in SignalV2: 

https://github.com/beacon-biosignals/Onda.jl/blob/1c69482315700342c706c1c4d4800231c0bc93e9/src/signals.jl#L80-L83

File: OndaEDFSchemas.jl/src/OndaEDFSchemas.jl [hunk 4]
```diff
@@ -1,94 +1,135 @@
...
+        kind::Union{Missing,AbstractString}
+        channel::Union{Missing,AbstractString}
+        sample_unit::Union{Missing,AbstractString}
+        sample_resolution_in_unit::Union{Missing,LPCM_SAMPLE_TYPE_UNION}
+        sample_offset_in_unit::Union{Missing,LPCM_SAMPLE_TYPE_UNION}
+        sample_type::Union{Missing,AbstractString}
+        sample_rate::Union{Missing,LPCM_SAMPLE_TYPE_UNION}
+        # errors, use `nothing` to indicate no error
+        error::Union{Nothing,String}
+    end
```
Comment (kleinschmidt) [inline OndaEDFSchemas.jl/src/OndaEDFSchemas.jl:84]: becuse the field names and types are different enough, might need to create two separate docstrings (maybe with some of hte common stuff templated out?)

File: OndaEDFSchemas.jl/src/OndaEDFSchemas.jl [hunk 5]
```diff
@@ -1,94 +1,135 @@
...
+    physical_minimum::Float32
+    physical_maximum::Float32
+    digital_minimum::Float32
+    digital_maximum::Float32
+    prefilter::String
+    samples_per_record::Int16
+    # EDF.FileHeader field
+    seconds_per_record::Float64
+    # Onda.SamplesInfo fields (channels -> channel), may be missing
+    sensor_type::Union{Missing,AbstractString} = lift(String, sensor_type)
```
Comment (kleinschmidt) [inline OndaEDFSchemas.jl/src/OndaEDFSchemas.jl:49]: also need `sensor_label`:

https://github.com/beacon-biosignals/Onda.jl/blob/1c69482315700342c706c1c4d4800231c0bc93e9/src/signals.jl#L121-L130

**Review by kleinschmidt — CHANGES_REQUESTED** [review]
good start!  the serialization depth thing is concerning to me, and I dont' really know what might be causing that now that wasn't before.  either something in legolas 0.5 has changed the serialization behavior, or arrow/arrowtypes has had an upgrade that caused some breakage there, oorrrrr something else more mysterious.

otherwise, some of the V2 fields need tweaking I think.  will likely need ot `using` more stuff from Onda for validating these fields (see the linked Onda source).

last thing is that I think we need to #55 first/include it in here, since `Onda.AnnotationV1` may now just drop the extra `value` field we create during import.

File: OndaEDFSchemas.jl/src/OndaEDFSchemas.jl (hunk 4 above)
Comment (christopher-dG) [reply OndaEDFSchemas.jl/src/OndaEDFSchemas.jl:84]: oops, yes definitely

File: OndaEDFSchemas.jl/test/runtests.jl (hunk 1 above)
Comment (christopher-dG) [reply OndaEDFSchemas.jl/test/runtests.jl:65]: @kleinschmidt not entirely sure what you mean by checking CI logs. The last run was [3 months ago](https://github.com/beacon-biosignals/OndaEDF.jl/actions/runs/3093327014/jobs/5005573529) on Onda 0.14 / Legolas 0.4 so I don't know that that tells us anything

Comment (kleinschmidt) [reply OndaEDFSchemas.jl/test/runtests.jl:65]: the only possibly relevant difference I can see is that that run was on Arrow 2.3.1 and we're getting 2.4 now

**Review by kleinschmidt — APPROVED** [review]

**Pull request merged**